CHANGES.md

```diff
@@ -1,18 +1,29 @@
 ringpop-go changes
 ==================
 
-v0.DEV (to be released)
------------------------
+v0.4.0
+------
 
-* Remove `File` and `Host`-based discover providers in favor of
-  `DiscoverProvider` interface #119
-* Ringpop will always assume the current host (`ringpop.node.identity`) is part
-  of the cluster. Previously, it was true for only `Host`-based discovery.
-* Add Go 1.6 testing on CI
+* Feature: Faulty nodes are now automatically reaped from nodes' membership
+  lists after 24 hours (by default). #123
+* New options for controlling suspect and reaping times. #123
+* Add new `Ready` and `Destroyed` events to ringpop. #125
+* Add additional logging to bring ringpop-go on par with ringpop-node log
+  messages. #116
+* Fix bug where Ringpop automatically added itself to the bootstrap hosts for
+  host-based bootstrapping, but not other bootstrapping methods. #120
+* Fix race condition where membership and hashring could be inconsistent with
+  each other. #112
+* Remove `File` and `Host` options from bootstrap options in favor of
+  `DiscoverProvider` interface. #120
+* Add Go 1.6 to testing on CI
```

Comment:
I think you're right. We could go into a lot more detail about every change, if we wanted. I think this turns the release into a bigger task.

Let me try to address each of your feedback items.

> This is changing behavior, not reflected in release notes (e.g. users might need to change their tooling).

Right. If the customer has created tooling that relies on faulty members being in the membership lists longer than 24 hours, then yes. Isn't that obvious enough from the one-line description, though?

The next line then suggests that the value can be tweaked.

> Another sentence on what you can do with them. E.g. "these are optional events that can invoke a user callback when ringpop is ready and destroyed, respectively"

I think "events" implies that. And at least, if you're a Ringpop user, you know what an event is and does. So I don't think any more information is necessary here.

Perhaps we could elaborate more (and give examples, etc) in help documentation.

> Nice, bug fix! But wait if I relied on self being added, what do I do now?
> Referenced PR (#120) references 3cc9e19, which has the change commented in more detail in CHANGES.md. Do you think the information is unnecessary?

Fair criticism; apologies that I removed that detail. I was trying to make the changelog easier to parse.

I think if people are relying on the behaviour of a bug, that is unfortunate. I don't know if the details of the bug should be in the changelog/release notes, though.